spec/datadog/profiling/stack_recorder_spec.rb

```diff
@@ -0,0 +1,78 @@
+# typed: ignore
+
+require 'datadog/profiling/spec_helper'
+require 'datadog/profiling/stack_recorder'
+
+RSpec.describe Datadog::Profiling::StackRecorder do
+  before { skip_if_profiling_not_supported(self) }
+
+  subject(:stack_recorder) { described_class.new }
+
+  # NOTE: A lot of libddprof integration behaviors are tested in the Collectors::Stack specs, since we need actual
+  # samples in order to observe what comes out of libddprof
```

Comment:
Note: This comment is accurate, but `Collectors::Stack` is only introduced in a later PR (#1966)